python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1499,15 +1513,25 @@ def do_evaluate(
         df: DataFrame,
     ) -> DataFrame:
         """Evaluate and return a dataframe."""
-        keys = broadcast(*(k.evaluate(df) for k in keys_in), target_length=df.num_rows)
+        # The scalars in preceding and following are constructed on the
```

Comment:
See https://github.com/rapidsai/cudf/pull/20291/files#r2437563679 for context.

Reply:
Or, as noted for the conditionaljoin case, we can construct in `__init__` and sync.